tests/upgrades/test_bookmarks.py

```diff
@@ -21,136 +21,137 @@
+    def test_post_create_public_disable_bookmark(self, dependent_scenario_name):
+        """Check the status of public disabled bookmark for all the
+        system entities(activation keys, tasks, compute profile, content hosts etc) after upgrade.
 
-    :expectedresults: Public enabled bookmark should be created successfully
+        :id: postupgrade-3b3abb85-cad2-4cbb-ad21-2780523351fd
 
-    :BZ: 1833264, 1826734, 1862119
-
-    :CaseImportance: Critical
-    """
+        :Steps:
 
-    for entity in BOOKMARK_ENTITIES:
-        book_mark_name = entity["name"] + request.node.name
-        bm = entities.Bookmark(
-            controller=entity['controller'],
-            name=book_mark_name,
-            public=True,
-            query=f"name={book_mark_name}",
-        ).create()
-        assert bm.controller == entity['controller']
-        assert bm.name == book_mark_name
-        assert bm.query == f"name={book_mark_name}"
-        assert bm.public
+            1. Check the bookmark status after post-upgrade.
+            2. Remove the bookmark.
 
+        :expectedresults: Public disabled bookmarks details for all the system entities
+        should be unchanged after upgrade.
 
-@post_upgrade(depend_on=test_pre_create_public_enable_bookmark)
-def test_post_create_public_enable_bookmark(request):
-    """Check the status of public enabled bookmark for all the
-    system entities(activation keys, tasks, compute profile, content hosts etc) after upgrade
+        :CaseImportance: Critical
+        """
+        pre_test_name = dependent_scenario_name
+        for entity in BOOKMARK_ENTITIES:
+            book_mark_name = entity["name"] + pre_test_name
+            bm = entities.Bookmark().search(query={'search': f'name="{book_mark_name}"'})[0]
+            assert bm.controller == entity['controller']
+            assert bm.name == book_mark_name
+            assert bm.query == f"name={book_mark_name}"
+            assert not bm.public
 
-    :id: 3b3abb85-cad2-4cbb-ad21-2780523351fd
+            bm.delete()
 
-    :Steps:
 
-        1. Check the bookmark status after post-upgrade
-        2. Remove the bookmark
-
-    :expectedresults: Public disabled bookmarks details for all the system entities
-    should be unchanged after upgrade
-
-    :CaseImportance: Critical
+class TestPublicEnableBookmark:
     """
-    pre_test_name = [
-        mark.kwargs['depend_on'].__name__
-        for mark in request.node.own_markers
-        if 'depend_on' in mark.kwargs
-    ][0]
-    for entity in BOOKMARK_ENTITIES:
-        book_mark_name = entity["name"] + pre_test_name
-        bm = entities.Bookmark().search(query={'search': f'name="{book_mark_name}"'})[0]
-        assert bm.controller == entity['controller']
-        assert bm.name == book_mark_name
-        assert bm.query == f"name={book_mark_name}"
-        assert bm.public
-        bm.delete()
+    Created Public enabled Bookmarks in pre upgrade should be unchanged after upgrade.
```

Comment:
@jyejare Is there any impact on the execution due to these comments?

Reply:
Fixed.

Reply:
@devendra104 No there is no impact in execution obviously but the doc-strings should be grammatically correct so that a new person working on it will easily understand it.